File: app/src/main/java/com/keno/solo/game_logic/Game.java
package com.keno.solo.game_logic;

import java.util.ArrayList;
import java.util.List;

public abstract class Game 
{
    private String name, rating, type, description;
    private ArrayList<String> genres;
    private int price;

    public Game
    (
        String name,
        String rating,
        String type,
        String description,
        List<String> genres,
        int price
    )
    {
        this.name = name;
        this.rating = rating;
        this.type = type;
        this.description = description;
        this.genres = new ArrayList<>(genres);
    }

    public String getName(){return name;}
    public String getRating(){return rating;}
    public String getType(){return type;}
    public String getDescription(){return description;}

    public ArrayList<String> getGenres(){return genres;}

    public int getPrice(){return price;}
}

File: sources/game_engine/Game.cpp
#include "game_engine/Game.hpp"

Game::Game
(
        std::string name,
        std::string rating,
        std::string type,
        std::string description,
        std::vector<std::string> genres,
        int price
)
{
        this->name = name;
        this->rating = rating;
        this->type = type;
        this->description = description;
        this->genres = genres;
        this->price = price;
}

std::string Game::getName() const {return name;}
std::string Game::getRating() const {return rating;}
std::string Game::getType() const {return type;}
std::string Game::getDescription() const {return description;}
std::vector<std::string> Game::getGenres() const {return genres;}
int Game::getPrice() const {return price;}